snippets/chainAsync.md

```diff
@@ -6,8 +6,11 @@ Loop through an array of functions containing asynchronous events, calling `next
 
 ```js
 const chainAsync = fns => {
-  let curr = 0;
-  const next = () => fns[curr++](next);
+  let curr = 0, last = fns[fns.length - 1];
+  const next = () => {
+    const fn = fns[curr++]
+    fn(fn === last ? null : next)
```

Comment:
This doesn't work like you'd think it does. It passed a null to the function. You should pass an undefined(for proper undefined checks) or modify this line to either call then function with next, or just call the function.

```suggestion
    fn === last ? fn() : fn(next);
```

Note on passing `undefined` versus a blank call(`()`):

I'd have to dig into the documentation to see if this would cause any issues with the `arguments` magic pointer(length of 0 versus length of k)

Reply:
although that'd be a pretty easy check, I'm currently swamped with papers

Reply:
```
function foo(){
  console.log(arguments.length)
}
foo()
// 0
foo(undefined)
// 1
```

Reply:
To keep the argument length consistent, I'll go with `undefined`.

Reply:
actually to keep it consistent, you should go with no arguments since that means otherwise you are passing something to it, so any code that relies on `arguments.length === 0` will fail for not receiving props on any possible overloading code that accepts differing lengths of arguments and checks, and even worse this could cause problems with auto curried closures that rely on the arguments length as well. You should call it without arguments

Reply:
Haha, fair enough.